client/src/pages/ScanStore.tsx

```diff
@@ -79,22 +101,47 @@ function ScanStore() {
   };
 
   const setDebugItem = (barcode: string) => {
-    setShoppingList(cartItems);
     setCurBarcode(barcode);
+    setShoppingList(cartItems);
   };
 
-  const addItem = () => {
-    let barcodes: string[] = [curBarcode];
-    fetchItem(barcodes);
+  const addItem = async () => {
+    if (curBarcode == "") {
+      // If empty, use media API
+      const imgReq = {
+        allowedMimeTypes: ["image/jpeg"],
+        allowedSources: ["camera"], // Restrict to camera scanning only
+      };
+      const imgRes = await microapps.requestMedia(imgReq);
+      setUploadImg(imgRes);
+    } else {
+      let barcodes: string[] = [curBarcode];
+      fetchItem(barcodes);
+    }
   };
 
   const fetchItem = async (barcodes: string[]) => {
-    let data = {
-      "merchant-id": merchantID,
-      barcode: barcodes,
-    };
-    const items = await fetchJson("POST", data, ITEM_LIST_API);
-    displayItems(items);
+    // Check first if we have previously retrieved this item
+    let unknown_barcodes: string[] = [];
+    let matchedItems: Item[] = [];
+    for (let i = 0; i < barcodes.length; ++i) {
+      if (idxMap[barcodes[i]] != undefined) {
+        let matchedCartItem = cartItems[idxMap[barcodes[i]]];
+        matchedItems.push(matchedCartItem.item);
+      } else {
+        unknown_barcodes.push(barcodes[i]);
+      }
+    }
+    if (matchedItems.length > 0) {
```

Comment:
nit: Replace with matchedItems.isNotEmpty (not sure whether tsx has this method for list, but if has, try to use it instead of checking the size.

Reply:
outdated. #61 fixed issue